TELEMETRY.md

```diff
@@ -0,0 +1,43 @@
+# Telemetry in the CodeQL extension for VS Code
+
+If you specifically opt-in to permit GitHub to do so, GitHub will collect usage data and metrics for the purposes of helping the core developers to improve the CodeQL extension for VS Code. This data will not be shared with any parties outside of GitHub and will be retained for a maximum of 30 days.
+
+## Why do you collect data?
+
+GitHub collects usage data and metrics to help us improve CodeQL for VS Code.
+
+## What data is collected
+
+GitHub collects anonymized information related to the usage of the extensions. The data collected are:
+
+- Which commands are run.
+- The time taken for each command.
+- Anonymous GUID to uniquely identify an installation.
+- IP address of the client sending the telemetry data.
+
+## How long will data be retained?
+
+IP address and GUIDs will be retained for a maximum of 30 days. Aggregated data of command identifiers and run times will be retained for a maximum of 180 days.
+
+## Who will have access to this data?
+
+IP address and GUIDs will only be available to the core developers of CodeQL. Aggregated data will be available to GitHub employees.
+
+## What data is **NOT** collected?
+
+We only collect the minimal amount of data we need to answer the questions about how our users are experiencing this product. To that end, we do not collect the following information:
+
+- No GitHub user ID
+- No CodeQL database names
+- No queries or anything about them
+- No filesystem paths.
+
+## How do I disable telemetry reporting?
+
+You can disable telemetry reporting by setting `codeQL.telemetry.enableTelemetry` to `false` in [your settings](https://code.visualstudio.com/docs/getstarted/settings#_settings-editor).
```

Comment:
Here it is worth repeating again that telemetry reporting is disabled until and unless the user explicitly agrees with it.